r/src/arrow_types.h

```diff
@@ -47,18 +47,8 @@ struct data {
     if (!(TEST)) Rcpp::stop(MSG); \
   } while (0)
 
-#define STOP_IF_NOT_OK(s) STOP_IF_NOT(s.ok(), s.ToString())
```

Comment:
The error is that `s` is evaluated **twice** if an error is encountered. Can be catastrophic if `s` has side-effects. It is also susceptible to operator precedence since it's not protected by parenthesis.

Reply:
👍 now that these macros are just aliases for functions, we could remove them right? (Not trying to increase your scope here, just for future consideration)

Reply:
Correct, I opted to minimize the diff while preserving code look.